### PR TITLE
Enumerator::Lazy#take should switch size with Infinity

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: kernel/common/enumerator.rb [hunk 1]
```diff
@@ -258,7 +258,7 @@ def take(n)
         raise ArgumentError, "attempt to take negative size" if n < 0
 
         current_size = enumerator_size
-        set_size = if current_size.kind_of?(Integer)
+        set_size = if current_size.kind_of?(Numeric)
```
Comment (YorickPeterse) [inline kernel/common/enumerator.rb]: Not sure if I understand this change, is `current_size` a Float in this case?

Comment (kachick) [inline kernel/common/enumerator.rb]: Yes, Infinity(Float::Infinity) is a Float.

```ruby
loop.size.class #=> Float
```

Comment (YorickPeterse) [inline kernel/common/enumerator.rb]: Ah yeah, now it makes sense.

**Comment by YorickPeterse** [comment]
Thanks!

**Comment by kachick** [comment]
Thanks to merge!

**Pull request merged**